### ISSUE TITLE
error in federated learning with websockets 

### ISSUE THREAD

**Issue by 1000ping** [OpenMined/PySyft#2390]
errors while trying to make websocket client workers in
Federated learning with websockets and federated averaging notebook in websockets-example-MNIST advanced tutorials
while executing this line 
after launching websocket servers on different terminals using 
python run_websocket_server.py --id alice --port 8887 
python run_websocket_server.py --id bob--port 8889 
python run_websocket_server.py --id charlie--port 8890

```
hook = sy.TorchHook(torch)

kwargs_websocket = {"host": "localhost", "hook": hook, "verbose": args.verbose}
alice = WebsocketClientWorker(id="alice", port=8887, **kwargs_websocket)
bob = WebsocketClientWorker(id="bob", port=8889, **kwargs_websocket)
charlie = WebsocketClientWorker(id="charlie", port=8890, **kwargs_websocket)

workers = [alice, bob, charlie]
print(workers)
```
.

> AttributeError                            Traceback (most recent call last)
> <ipython-input-10-dfd3f30b67a3> in <module>
>       2 
>       3 kwargs_websocket = {"host": "localhost", "hook": hook, "verbose": args.verbose}
> ----> 4 alice = WebsocketClientWorker(id="alice", port=8887, **kwargs_websocket)
>       5 bob = WebsocketClientWorker(id="bob", port=8889, **kwargs_websocket)
>       6 charlie = WebsocketClientWorker(id="charlie", port=8890, **kwargs_websocket)
> 
> h:\softwares\anaconda\envs\dlpytorch\lib\site-packages\syft\workers\websocket_client.py in __init__(self, hook, host, port, secure, id, is_client_worker, log_msgs, verbose, data)
>      49         self.secure = secure
>      50         self.ws = None
> ---> 51         self.connect()
>      52 
>      53     @property
> 
> h:\softwares\anaconda\envs\dlpytorch\lib\site-packages\syft\workers\websocket_client.py in connect(self)
>      61             args["sslopt"] = {"cert_reqs": ssl.CERT_NONE}
>      62 
> ---> 63         self.ws = websocket.create_connection(**args)
>      64 
>      65     def close(self):
> 
> AttributeError: module 'websocket' has no attribute 'create_connection'

and after I changed
`import websocket`
with
`import websocket._core as websocket`
in syft/workers/websocket-client.py
after executing same notebook line 
in am getting

> OverflowError                             Traceback (most recent call last)
> <ipython-input-5-3a01ed8c63d6> in <module>
>       2 
>       3 kwargs_websocket = {"host": "localhost", "hook": hook, "verbose": args.verbose}
> ----> 4 alice = WebsocketClientWorker(id="alice", port=8887, **kwargs_websocket)
>       5 bob = WebsocketClientWorker(id="bob", port=8889, **kwargs_websocket)
>       6 charlie = WebsocketClientWorker(id="charlie", port=8890, **kwargs_websocket)
> 
> h:\softwares\anaconda\envs\dlpytorch\lib\site-packages\syft\workers\websocket_client.py in __init__(self, hook, host, port, secure, id, is_client_worker, log_msgs, verbose, data)
>      49         self.secure = secure
>      50         self.ws = None
> ---> 51         self.connect()
>      52 
>      53     @property
> 
> h:\softwares\anaconda\envs\dlpytorch\lib\site-packages\syft\workers\websocket_client.py in connect(self)
>      61             args["sslopt"] = {"cert_reqs": ssl.CERT_NONE}
>      62 
> ---> 63         self.ws = websocket.create_connection(**args)
>      64 
>      65     def close(self):
> 
> h:\softwares\anaconda\envs\dlpytorch\lib\site-packages\websocket\_core.py in create_connection(url, timeout, class_, **options)
>     512                      skip_utf8_validation=skip_utf8_validation, **options)
>     513     websock.settimeout(timeout if timeout is not None else getdefaulttimeout())
> --> 514     websock.connect(url, **options)
>     515     return websock
> 
> h:\softwares\anaconda\envs\dlpytorch\lib\site-packages\websocket\_core.py in connect(self, url, **options)
>     221         self.sock_opt.timeout = options.get('timeout', self.sock_opt.timeout)
>     222         self.sock, addrs = connect(url, self.sock_opt, proxy_info(**options),
> --> 223                                    options.pop('socket', None))
>     224 
>     225         try:
> 
> h:\softwares\anaconda\envs\dlpytorch\lib\site-packages\websocket\_http.py in connect(url, options, proxy, socket)
>     118     sock = None
>     119     try:
> --> 120         sock = _open_socket(addrinfo_list, options.sockopt, options.timeout)
>     121         if need_tunnel:
>     122             sock = _tunnel(sock, hostname, port, auth)
> 
> h:\softwares\anaconda\envs\dlpytorch\lib\site-packages\websocket\_http.py in _open_socket(addrinfo_list, sockopt, timeout)
>     160         family, socktype, proto = addrinfo[:3]
>     161         sock = socket.socket(family, socktype, proto)
> --> 162         sock.settimeout(timeout)
>     163         for opts in DEFAULT_SOCKET_OPTION:
>     164             sock.setsockopt(*opts)
> 
> OverflowError: timeout doesn't fit into C timeval

OS version windows 10 64bit
python version 3.6.8
Cuda 10

Installed python libraries
 Name                    Version                   Build  Channel
absl-py                   0.7.1                    pypi_0    pypi
astor                     0.8.0                    pypi_0    pypi
attrs                     19.1.0                   pypi_0    pypi
backcall                  0.1.0                    pypi_0    pypi
bleach                    3.1.0                    pypi_0    pypi
certifi                   2019.3.9                 py36_0
cffi                      1.12.3                   pypi_0    pypi
click                     7.0                      pypi_0    pypi
colorama                  0.4.1                    pypi_0    pypi
cycler                    0.10.0                   pypi_0    pypi
decorator                 4.4.0                    pypi_0    pypi
defusedxml                0.6.0                    pypi_0    pypi
dill                      0.3.0                    pypi_0    pypi
entrypoints               0.3                      pypi_0    pypi
enum34                    1.1.6                    pypi_0    pypi
flask                     1.1.1                    pypi_0    pypi
flask-socketio            4.1.0                    pypi_0    pypi
gast                      0.2.2                    pypi_0    pypi
gevent                    1.4.0                    pypi_0    pypi
google-pasta              0.1.7                    pypi_0    pypi
greenlet                  0.4.15                   pypi_0    pypi
grpcio                    1.22.0                   pypi_0    pypi
h5py                      2.9.0                    pypi_0    pypi
ipykernel                 5.1.1                    pypi_0    pypi
ipython                   7.6.1                    pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
itsdangerous              1.1.0                    pypi_0    pypi
jedi                      0.14.1                   pypi_0    pypi
jinja2                    2.10.1                   pypi_0    pypi
joblib                    0.13.2                   pypi_0    pypi
json5                     0.8.5                    pypi_0    pypi
jsonschema                3.0.1                    pypi_0    pypi
jupyter-client            5.3.1                    pypi_0    pypi
jupyter-core              4.5.0                    pypi_0    pypi
jupyter-http-over-ws      0.0.6                    pypi_0    pypi
jupyterlab                1.0.2                    pypi_0    pypi
jupyterlab-server         1.0.0                    pypi_0    pypi
keras-applications        1.0.8                    pypi_0    pypi
keras-preprocessing       1.1.0                    pypi_0    pypi
kiwisolver                1.1.0                    pypi_0    pypi
lz4                       2.1.10                   pypi_0    pypi
markdown                  3.1.1                    pypi_0    pypi
markupsafe                1.1.1                    pypi_0    pypi
matplotlib                3.1.1                    pypi_0    pypi
mistune                   0.8.4                    pypi_0    pypi
msgpack                   0.6.1                    pypi_0    pypi
nbconvert                 5.5.0                    pypi_0    pypi
nbformat                  4.4.0                    pypi_0    pypi
notebook                  5.7.8                    pypi_0    pypi
numpy                     1.16.4                   pypi_0    pypi
pandocfilters             1.4.2                    pypi_0    pypi
parso                     0.5.1                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    6.0.0                    pypi_0    pypi
pip                       19.1.1                   py36_0
prometheus-client         0.7.1                    pypi_0    pypi
prompt-toolkit            2.0.9                    pypi_0    pypi
protobuf                  3.8.0                    pypi_0    pypi
pycparser                 2.19                     pypi_0    pypi
pygments                  2.4.2                    pypi_0    pypi
pyparsing                 2.4.0                    pypi_0    pypi
pyrsistent                0.15.3                   pypi_0    pypi
python                    3.6.8                h9f7ef89_7
python-dateutil           2.8.0                    pypi_0    pypi
python-engineio           3.8.2.post1              pypi_0    pypi
python-socketio           4.2.0                    pypi_0    pypi
pywinpty                  0.5.5                    pypi_0    pypi
pyyaml                    5.1.1                    pypi_0    pypi
pyzmq                     18.0.2                   pypi_0    pypi
scikit-learn              0.21.2                   pypi_0    pypi
scipy                     1.3.0                    pypi_0    pypi
send2trash                1.5.0                    pypi_0    pypi
setuptools                41.0.1                   py36_0
six                       1.12.0                   pypi_0    pypi
sqlite                    3.28.0               he774522_0
syft                      0.1.21a1                 pypi_0    pypi
tblib                     1.4.0                    pypi_0    pypi
tensorboard               1.14.0                   pypi_0    pypi
tensorflow                1.14.0                   pypi_0    pypi
tensorflow-estimator      1.14.0                   pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
terminado                 0.8.2                    pypi_0    pypi
testpath                  0.4.2                    pypi_0    pypi
tf-encrypted              0.5.4                    pypi_0    pypi
torch                     1.1.0                    pypi_0    pypi
torchsummary              1.5.1                    pypi_0    pypi
torchvision               0.3.0                    pypi_0    pypi
tornado                   6.0.3                    pypi_0    pypi
traitlets                 4.3.2                    pypi_0    pypi
vc                        14.1                 h0510ff6_4
vs2015_runtime            14.15.26706          h3a45250_4
wcwidth                   0.1.7                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
websocket                 0.2.1                    pypi_0    pypi
websocket-client          0.56.0                   pypi_0    pypi
websockets                8.0.1                    pypi_0    pypi
werkzeug                  0.15.4                   pypi_0    pypi
wheel                     0.33.4                   py36_0
wincertstore              0.2              py36h7fe50ca_0
wrapt                     1.11.2                   pypi_0    pypi
zstd                      1.4.1.0                  pypi_0    pypi

**Comment by kakirastern** [comment]
I tried substituting for the L63:  `self.ws = websocket.create_connection(**args)` with the following and it seemed to have worked to some degree: 

```
self.ws = websockets.connect(uri=url, **args)
```

But it also seems like in this module to accompany the `import websockets` you will probably also need to do `import asyncio`, which is currently missing from it. I am working on the Websocket Workers tutorials too so I encountered some similar issues. Hope that they will get resolved really soon.
